triedb/pathdb/history_index.go

```diff
@@ -106,38 +114,20 @@ func (r *indexReader) refresh() error {
 	// may have been modified by additional elements written to the disk.
 	if len(r.descList) != 0 {
 		last := r.descList[len(r.descList)-1]
-		if !last.full() {
```

Comment:
It's very unlikely that the last reader is full.

By removing it and we always need to re-open the last reader for the potential new index element. Performance wise it's basically no difference.